gcovr/__main__.py

```diff
@@ -152,6 +152,10 @@ def main(args=None):
 
     logger = Logger(options.verbose)
 
+    if sys.version_info < (3, 8) and options.use_canonical_paths:
+        logger.warn("--use_canonical_paths will be ignored due to incompatible Python version.")
+        options.use_canonical_paths = None
```

Comment:
The user explicite requested the canonical paths. I'll prefer an error and exit if the option is used instead of ignoring it.

Reply:
Ok, will change that.